DtMaintenance/Data/Repository/EquipmentDbRepository.cs

```diff
@@ -1,12 +1,56 @@
 ﻿using DumptruckMaintenance.Data.Db;
 using DumptruckMaintenance.Domain.Entity;
+using DumptruckMaintenance.Common.Filters;
+using System.Collections.Generic;
 
 namespace DumptruckMaintenance.Data.Repository {
-    public class EquipmentDbRepository : DbRepository<Equipment> { 
+    public class EquipmentDbRepository : DbRepository<Equipment> {
         public EquipmentDbRepository(Database database) : base(database) { }
 
         public Equipment GetBySerialNumber(string serialNumber) {
             return Get("SerialNumber", serialNumber);
         }
+
+        public IList<Equipment> FilterBy(EquipmentFilter filter) {
+            using (var session = Source.OpenSession()) {
+                using (session.BeginTransaction()) {
+                    Equipment equipment = null;
+                    EquipmentType equipmentType = null;
+                    Rfid rfid = null;
+                    var query = session.QueryOver(() => equipment)
+                        .JoinAlias(() => equipment.Type, () => equipmentType)
+                        .JoinAlias(() => equipment.Rfid, () => rfid);
+                    if (!string.IsNullOrWhiteSpace(filter.RfidKey)) {
+                        query
+                            .WhereRestrictionOn(() => rfid.Key)
+                            .IsInsensitiveLike($"%{filter.RfidKey}%");
+                    }
+                    if (!string.IsNullOrWhiteSpace(filter.TypeName)) {
+                        query
+                            .WhereRestrictionOn(() => equipmentType.Name)
+                            .IsInsensitiveLike($"%{filter.TypeName}%");
+                    }
+                    if (!string.IsNullOrWhiteSpace(filter.SerialNumber)) {
+                        query
+                            .WhereRestrictionOn(() => equipment.SerialNumber)
+                            .IsInsensitiveLike($"%{filter.SerialNumber}%");
+                    }
+                    if (filter.CommissioningDate != null) {
+                        query.Where(() => equipment.CommissioningDate.Value.Year == filter.CommissioningDate.Value.Year);
```

Comment:
Может быть стоит разбить строку как-нибудь вот так, чтобы соответствовало code style'у?
```suggestion
                        query.Where(
                            e => e.CommissioningDate.Value.Year == filter.CommissioningDate.Value.Year
                        );